k4FWCore/include/k4FWCore/FunctionalUtils.h

```diff
@@ -136,15 +139,35 @@ namespace k4FWCore {
             if constexpr (std::is_same_v<EDM4hepType, const podio::CollectionBase*>) {
               inputMap.push_back(&get(handle, thisClass, Gaudi::Hive::currentContext()));
             } else {
-              podio::CollectionBase* in = handle.get()->get();
-              inputMap.push_back(static_cast<EDM4hepType*>(in));
+              podio::CollectionBase* in      = handle.get()->get();
+              auto*                  typedIn = dynamic_cast<EDM4hepType*>(in);
+              if (typedIn) {
+                inputMap.push_back(static_cast<EDM4hepType*>(typedIn));
```

Comment:
```suggestion
                inputMap.push_back(typedIn);
```
No need to cast again, this already has the correct type.